pkg/forwarder/forwarder.go

```diff
@@ -148,10 +150,20 @@ func New(cfg Config) (*Worker, error) {
 
 	// Create the `fromClient`.
 	fromTransport := metricsclient.DefaultTransport()
+	fromClient := &http.Client{Transport: fromTransport}
+
 	if len(cfg.FromCAFile) > 0 {
-		if fromTransport.TLSClientConfig == nil {
-			fromTransport.TLSClientConfig = &tls.Config{}
+		level.Info(logger).Log("msg", "fromClient transport: CA file %s and TLS", cfg.FromCAFile)
+
+		var cert tls.Certificate
+		var err error
```

Comment:
to be safe in case metricsclient.DefaultTransport() happens to initialize the TLSClientConfig in the future.
```suggestion
		var err error

		if fromTransport.TLSClientConfig == nil {
			fromTransport.TLSClientConfig = &tls.Config{}
		}
```